### DISCUSSION TITLE
Feature Request: Expand Citations to Display Full Original Document Content

### DISCUSSION THREAD

**Discussion by YuMaiwen** [langgenius/dify#3169] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

Currently, when users click on citations within a discussion thread, the system fetches and displays the excerpted content of the relevant paragraphs from the referenced knowledge source. 

### 2. Describe the feature you'd like to see

We would like to see a feature that, upon clicking a citation within a conversation, allows users to view the complete content of the original source document. 

### 3. How will this feature improve your workflow or experience?

Full Access: Users can explore topics more deeply and holistically without leaving the platform or separately searching for the document.
Multimedia Preservation: Original documents often include valuable visual aids (e.g., images, diagrams) lost in extracted segments.
Contextual Completeness: Access to the entire document ensures users have all necessary information for accurate interpretation.

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by simongaocd** [comment]
any update? need a feature with download the  complete original document

**Comment by r0kk** [comment]
Hi, first of all great work. I have one consideration regarding using dify and that is end user interaction with knowledge bases. **Ragflow** UX/UI is superb, so it would be great if there would be similar feature in dify. Also **kotaemon** user experience is quite great.

If there is a way to interact with knowledge bases already, I would be very grateful 🙏. Thank you.

**Comment by nickerber** [comment]
This would be a gamechanger. Having the possibility to see the source document(s), jump to the position the chunk has been taken from and maybe highlight the text to dive deeper into the information given. Need this!

**Comment by rnavarroz** [comment]
Is there any plan to include this feature?
